iot-hub/Samples/service/CleanupDevicesSample/CleanupDevicesSample.cs

```diff
@@ -0,0 +1,173 @@
+        private static readonly IReadOnlyList<JobStatus> s_completedJobs = new[]
+        {
+            JobStatus.Completed,
+            JobStatus.Failed,
+            JobStatus.Cancelled,
+        };
+
+        private readonly RegistryManager _registryManager;
+        private readonly BlobContainerClient _blobContainerClient;
+        private readonly List<string> _deleteDeviceWithPrefix;
+
+        public CleanupDevicesSample(RegistryManager rm, BlobContainerClient sc, List<string> deleteDeviceWithPrefix)
+        {
+            _registryManager = rm ?? throw new ArgumentNullException(nameof(rm));
+            _blobContainerClient = sc ?? throw new ArgumentNullException(nameof(sc));
+
+            _deleteDeviceWithPrefix = deleteDeviceWithPrefix;
+        }
+
+        public async Task RunCleanUpAsync()
+        {
+            // Clean up devices
+            await PrintDeviceCountAsync();
+            await CleanupDevices();
+        }
+
+        private async Task CleanupDevices()
+        {
+            Console.WriteLine($"Storage container {_blobContainerClient.Name} created.");
+
+            // Retrieve the SAS Uri that will be used to grant access to the storage containers.
+            string storageAccountSasUri = GetStorageAccountSasUri(_blobContainerClient).ToString();
+
+            // Step 1: Export all device identities.
+            JobProperties exportAllDevicesProperties = JobProperties
+                .CreateForExportJob(
+                    outputBlobContainerUri: storageAccountSasUri,
+                    excludeKeysInExport: true,
+                    storageAuthenticationType: StorageAuthenticationType.KeyBased);
+
+            JobProperties exportAllDevicesJob = await _registryManager.ExportDevicesAsync(exportAllDevicesProperties);
+
+            // Wait until the export job is finished.
+            while (true)
+            {
+                exportAllDevicesJob = await _registryManager.GetJobAsync(exportAllDevicesJob.JobId);
+                if (s_completedJobs.Contains(exportAllDevicesJob.Status))
+                {
+                    // Job has finished executing.
+                    break;
+                }
+
+                Console.WriteLine($"Job {exportAllDevicesJob.JobId} is {exportAllDevicesJob.Status} with progress {exportAllDevicesJob.Progress}%");
+                await Task.Delay(s_waitDuration);
+            }
+            Console.WriteLine($"Job {exportAllDevicesJob.JobId} is {exportAllDevicesJob.Status}.");
+
+            if (exportAllDevicesJob.Status != JobStatus.Completed)
+            {
+                throw new Exception("Exporting devices failed, exiting.");
+            }
+
+            // Step 2: Download the exported devices list.
+            BlobClient blobClient = _blobContainerClient.GetBlobClient(ImportExportDevicesFileName);
+            BlobDownloadInfo download = await blobClient.DownloadAsync();

```

Comment:
What this line does wasn't self explanatory to me and had to look up.

Either adding a comment something like "Download the previously created blob" or renaming will add some more context.